frontend/ui/widget/filechooser.lua

```diff
@@ -88,11 +89,13 @@ function FileChooser:show_dir(dirname)
     return true
 end
 
-function FileChooser:show_file(filename)
+function FileChooser:show_file(filename, fullpath)
     for _, pattern in ipairs(self.exclude_files) do
         if filename:match(pattern) then return false end
     end
-    return self.show_unsupported or self.file_filter == nil or self.file_filter(filename)
+    if not self.show_unsupported and self.file_filter ~= nil and not self.file_filter(filename) then return false end
+    if not self.show_finished and fullpath ~= nil and filemanagerutil.getStatus(fullpath) == "complete" then return false end
```

Comment:
I think `local FileChooser = Menu:extend{` might be missing:
```diff
    show_hidden = false, -- set to true to show folders/files starting with "."
+   show_finished = true, -- show all books
```
In case it's used from something else than FileManager.

Reply:
Yes, and more over.
If we start with, say, Last book, and call PathChooser, it doesn't follow our view settings, because they are read in FM:
https://github.com/koreader/koreader/blob/0ac258fcb70b113f6f699e51cd9b30825c22e18d/frontend/apps/filemanager/filemanager.lua#L132
I think all these 3 view flags must be initiated directly in `local FileChooser = Menu:extend{`.

Reply:
> won't be reflected if we change the setting

We change `self` when changing a setting
https://github.com/koreader/koreader/blob/0ac258fcb70b113f6f699e51cd9b30825c22e18d/frontend/ui/widget/filechooser.lua#L458-L459

Reply:
Yes, but this will apply to the current `self`, not to future FileChooser instances.

Reply:
Anyway, I was going to rewrite these three togglings a bit.
I think no need to have their parts in both FM and FileChooser, they can be combined in one place.
https://github.com/koreader/koreader/blob/0ac258fcb70b113f6f699e51cd9b30825c22e18d/frontend/apps/filemanager/filemanager.lua#L788-L791
https://github.com/koreader/koreader/blob/0ac258fcb70b113f6f699e51cd9b30825c22e18d/frontend/ui/widget/filechooser.lua#L458-L461

Reply:
> not to future FileChooser instances.

True, we have
https://github.com/koreader/koreader/blob/0ac258fcb70b113f6f699e51cd9b30825c22e18d/frontend/ui/widget/pathchooser.lua#L39

Reply:
And we do not need this in FM, because the setting can be changed within FM only.

Reply:
Or not, changing the setting doesn't change the base class. I'll think of it.

Reply:
You could change the setting in the class:
```diff
 function FileChooser:toggleUnsupportedFiles() 
-     self.show_unsupported = not self.show_unsupported 
+     FileChooser.show_unsupported = not FileChooser.show_unsupported 
```